### DISCUSSION TITLE
How to sort by "latest changed state"

### DISCUSSION THREAD

**Discussion by thefunkygibbon** [netalertx/NetAlertX#828] [General]
Hi,  having been used to using Fing, it has the ability to sort it based on the last state change, which is useful for troubleshooting and working out whats joined the network recently etc (or dropped off the network).  
Is there a way to sort this?  I see there is a "last offline" but i'm not sure that that would satisfy this use-case

thoughts?

**Comment by jokob-sk** [comment]
Hi, 

I think the Events section covers that. You can enter "Connected" in the search bar and that will display connected and disconnected events and the device:

![image](https://github.com/user-attachments/assets/7baed578-e181-443a-911b-ef6e02244265)

**Comment by thefunkygibbon** [comment]
ah yes, that seems to be what i'm looking for, thanks.   out of interest, what does "VOIDED - Disconnected" mean different to just a normal disconnected?

**Reply by jokob-sk** [reply]
I'm really not 100% sure. 

I've been avoiding this part of the app since I started maintaining it. The code isn't very readable IMO so I tried to avoid touching it. I know there are a couple of bugs, but I'd say they are non-breaking issues. Related discussion #814 

Relevant code if you want to try to figure it out yourself :) 

https://github.com/jokob-sk/NetAlertX/blob/main/front/php/server/events.php
https://github.com/jokob-sk/NetAlertX/blob/main/server/networkscan.py#L176

**Comment by thefunkygibbon** [comment]
from what I can tell from the code, 
"VOIDED" events are those that have been deemed incorrect or irrelevant, often due to false or temporary network issues like brief disconnections. In the script, events like "Connected" or "Disconnected" are voided if a disconnection occurs within 3 minutes of a connection for the same device, suggesting it was likely a momentary glitch. The voiding process helps maintain accurate event records by labeling these invalidated events as "VOIDED - " followed by the original event type, ensuring that alerts and data reflect only meaningful changes.

(note I can't understand code in the slightest, this was chatgpt's interpretation from looking at both of those python scripts ;-) )

**Reply by jokob-sk** [reply]
Haha there you go. Sounds a bit wonky but no one flagged this as a bug for now.
